### PR TITLE
Create durian executable and implement pretty printing in it

### PR DESCRIPTION
**Pull request by rwblickhan**
---

## :construction_worker: Changes

Added a single executable to run all Durian commands.

## :flashlight: Testing Instructions

Just do ./durian --help! Note that most of it is unimplemented but pretty printing should work.

### CODE REVIEW

**Comment by rwblickhan** [comment]
This won't build on Travis until #54 is closed (which changes Travis to use a more modern gcc that actually has C++14/17 stuff).

File: src/main.cpp [hunk 1]
```diff
@@ -1,5 +1,130 @@
...
+    void setModeIfExistsFlag(const std::string& flag, Mode mode) {
+        if (optionExists(flag)) {
+            if (this->mode != Mode::None) {
+                std::cout << "Cannot specify multiple modes!" << std::endl;
+                printHelpAndExit(1);
+            }
+            this->mode = mode;
+        }
+    }
+    [[noreturn]] void printHelpAndExit(int exitCode) {
```
Comment (bobheadxi) [inline src/main.cpp:57]: `[[noreturn]]`?

Comment (rwblickhan) [reply src/main.cpp:57]: It’s something clang-tidy recommended. It just means the function will *never* return which apparently enables some optimizations/warnings: https://stackoverflow.com/questions/10538291/what-is-the-point-of-noreturn

Comment (rwblickhan) [reply src/main.cpp:57]: Arguably it’s also useful as documentation; “yes, I know this function calls exit and closes the program”.

Comment (bobheadxi) [reply src/main.cpp:57]: wow thats neat

**Comment by rwblickhan** [comment]
I've rebased on the cleanup branch so that PR should be merged first (but at least it fixes the broken build 🤷‍♀️).

**Comment by rwblickhan** [comment]
... or not. I'll look into it more tomorrow.

**Review by bobheadxi — APPROVED** [review]
This is broken as fuck, but oh well, we will come back to this! Awesome work

**Pull request merged**